> probably some Windows + NumPy pre-release stuff a bit like #19804?

It's possible that something got stricter in pytest 8 of course, but it looks very different from the multiple-warnings issue in #19804:
```
      assert mod not in mods
  E   AssertionError
          mod        = <module 'scipy.conftest' from 'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\cibw-run-xjxg7tr2\\cp39-win_amd64\\venv-test\\lib\\site-packages\\scipy\\conftest.py'>
          mods       = [<module 'scipy.conftest' from 'C:\\Users\\runneradmin\\AppData\\Local\\Temp\\cibw-run-xjxg7tr2\\cp39-win_amd64\\venv-test\\lib\\site-packages\\scipy\\conftest.py'>]
```

These two are the same but apparently shouldn't be. I didn't see any obviously [related changes](https://github.com/pytest-dev/pytest/blame/8.0.0rc1/src/_pytest/config/__init__.py#L662) in pytest for that, but found that this is https://github.com/pytest-dev/pytest/issues/9765.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -130,7 +130,7 @@ skip = "cp36-* cp37-* cp38-* pp* *_ppc64le *_i686 *_s390x"
 build-verbosity = "3"
 # gmpy2 and scikit-umfpack are usually added for testing. However, there are
 # currently wheels missing that make the test script fail.
-test-requires = ["pytest", "pytest-cov", "pytest-xdist", "mpmath", "threadpoolctl", "pooch", "hypothesis"]
+test-requires = ["pytest==7.4.3", "pytest-cov", "pytest-xdist", "mpmath", "threadpoolctl", "pooch", "hypothesis"]
 test-command = "bash {project}/tools/wheels/cibw_test_command.sh {project}"
 
 [tool.cibuildwheel.linux]

